crates/biome_service/src/workspace/server.rs

```diff
@@ -851,3 +847,9 @@ impl Workspace for WorkspaceServer {
 fn is_dir(path: &Path) -> bool {
     path.is_dir() || (path.is_symlink() && fs::read_link(path).is_ok_and(|path| path.is_dir()))
 }
+
+fn make_search_pattern_id() -> String {
```

Comment:
What's this for?

Reply:
For searching, the workspace protocol is split into three functions: `parse_pattern()`, `search_pattern()`, and `drop_pattern()`. The reason for this is that we only need to parse once, and all workers can search using the pattern by referencing a handle instead of parsing again. This function generates the IDs for these "handles".